app/Models/Asset.php

```diff
@@ -112,7 +113,7 @@ public function declinedCheckout(User $declinedBy, $signature)
         'location_id'      => 'nullable|exists:locations,id',
         'rtd_location_id'  => 'nullable|exists:locations,id',
         'purchase_date'    => 'nullable|date|date_format:Y-m-d',
-        'serial'           => 'nullable|unique_undeleted:assets,serial',
+        'serial'           => 'unique_undeleted:assets,serial|required_serial',
```

Comment:
Nitpick that doesn't hold up merging: seeing `required_serial` without context might imply that `serial` is required but has some special rules around it. Calling it something like `potentially_required:serial` would be more accurate but requires updating the custom validator's method signature to account for the parameter `serial`. (which might make it dynamic for any future fields that follow this path).